flow/layers/opacity_layer.cc

```diff
@@ -8,10 +8,19 @@
 
 namespace flutter {
 
-OpacityLayer::OpacityLayer(int alpha, const SkPoint& offset)
-    : alpha_(alpha), offset_(offset) {}
+#if defined(OS_FUCHSIA)
+constexpr bool kRenderOpacityUsingSystemCompositor = true;
+#endif
 
-OpacityLayer::~OpacityLayer() = default;
+OpacityLayer::OpacityLayer(const SkPoint& offset, int alpha
+  #if defined(OS_FUCHSIA)
```

Comment:
@chinmaygarde : I think you mentioned that we're moving away from `#if defined(OS_FUCHSIA)` once the runner is migrated to Flutter? That is, the engine should not know which client is using the API. The engine should always provide the same (probably more general) API, and the client may choose how to use the API.

Reply:
I tried to do that in this CL, but it ended up becoming humongous.

My next CL (which optimizes SceneUpdateContext) will accomplish this

Reply:
So, I managed to get rid of a lot of OS_FUCHSIA by following this philosophy.

There are still several left, especially in SceneUpdateContext, but it's a lot better now.

It does mean there is the possibility of a non-Fuchsia "system-composited" path that would fail to render properly, but I added DCHECK to detect this case.  It would probably be possible to extend this path to other platforms in the future.